### PR TITLE
rgw: pass RGWProcessEnv into process_request()

### PR DESCRIPTION
**Pull request by cbodley**
shares `struct RGWProcessEnv` between frontends, instead of making a copy for each. this lets us use it to store globals and reload parts of it in `RGWRealmReloader` without frontend interaction. passes a `const RGWProcessEnv&` into `process_request()` for access in rgw ops. moves `struct RGWProcessEnv` into a new header rgw_process_env.h so ops don't have to include all of rgw_process.h

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by cbodley** [comment]
this work is related to https://github.com/ceph/ceph/pull/48482, but i split it out here. i plan to use this `struct RGWProcessEnv` to expose `rgw::sal::ConfigStore` and `rgw::SiteConfig` for use in rgw ops

File: src/rgw/rgw_asio_frontend.cc [hunk 1]
```diff
@@ -266,13 +266,8 @@ void handle_connection(boost::asio::io_context& context,
...
       const auto started = ceph::coarse_real_clock::now();
       ceph::coarse_real_clock::duration latency{};
-      process_request(env.store, env.rest, &req, uri_prefix,
-                      *env.auth_registry, &client, env.olog, y,
-                      scheduler, &user, &latency,
-                      env.ratelimiting->get_active(),
-                      env.lua_background,
-                      env.lua_manager,
-                      &http_ret);
+      process_request(env, &req, uri_prefix, &client, y,
```
Comment (mattbenjamin) [inline src/rgw/rgw_asio_frontend.cc:269]: ack

File: src/rgw/rgw_process.h [hunk 2]
```diff
@@ -155,20 +155,14 @@ class RGWLoadGenProcess : public RGWProcess {
   void set_access_key(RGWAccessKey& key) { access_key = key; }
 };
 /* process stream request */
-extern int process_request(rgw::sal::Store* store,
-                           RGWREST* rest,
+extern int process_request(const RGWProcessEnv& penv,
```
Comment (mattbenjamin) [inline src/rgw/rgw_process.h:142]: nice cleanup

**Review by mattbenjamin — APPROVED** [review]
this all looks good, to me

**Comment by cbodley** [comment]
thanks for review! still need to fix some compilation errors in unit tests

File: src/rgw/rgw_appmain.cc [hunk 3]
```diff
@@ -468,24 +460,25 @@ int rgw::AppMain::init_frontends2(RGWLib* rgwlib)
...
-  if (store->get_name() == "rados") {
+  if (env.store->get_name() == "rados") {
     // add a watcher to respond to realm configuration changes
-    pusher = std::make_unique<RGWPeriodPusher>(dpp, store, null_yield);
-    fe_pauser = std::make_unique<RGWFrontendPauser>(fes, *(implicit_tenant_context.get()), pusher.get());
+    pusher = std::make_unique<RGWPeriodPusher>(dpp, env.store, null_yield);
+    fe_pauser = std::make_unique<RGWFrontendPauser>(fes, pusher.get());
     rgw_pauser = std::make_unique<RGWPauser>();
     rgw_pauser->add_pauser(fe_pauser.get());
     if (lua_background) {
```
Comment (yuvalif) [inline src/rgw/rgw_appmain.cc:475]: since we store the lua background pointer inside `RGWProcessEnv` which is already a member of `AppMain`, maybe we don't need to also store it in `AppMain`, and use the `unique_ptr` in the env?

Comment (yuvalif) [reply src/rgw/rgw_appmain.cc:475]: @cbodley ?

File: src/rgw/rgw_appmain.cc [hunk 4]
```diff
@@ -536,6 +535,8 @@ void rgw::AppMain::init_lua()
   }
 #endif
 
+  env.lua.manager = env.store->get_lua_manager();
```
Comment (yuvalif) [inline src/rgw/rgw_appmain.cc:537]: why was this moved from line 411 to here?

Comment (cbodley) [reply src/rgw/rgw_appmain.cc:537]: i just thought that `init_lua()` was a better place than `init_frontends2()`

Comment (yuvalif) [reply src/rgw/rgw_appmain.cc:537]: this needs to be tested with zone reloads.
the lua manager is used to access the system objects.

**Comment by cbodley** [comment]
failed qa in https://pulpito.ceph.com/cbodley-2022-12-12_18:46:21-rgw-wip-cbodley-testing-distro-default-smithi/ with some ceph_test_librgw_file crashes. should be fixed now

**Comment by cbodley** [comment]
passed qa in https://pulpito.ceph.com/cbodley-2022-12-14_14:28:42-rgw-wip-cbodley-testing-distro-default-smithi/
